src/transformers/modeling_tf_utils.py

```diff
@@ -962,6 +962,153 @@ def init_copy_embeddings(old_embeddings, new_num_tokens):
     return mask, current_weights
 
 
+class TFNestLayer(base_layer.Layer):
+    """
+    Custom keras layer that subclasses tf.keras.layers.Layer to enable access to all sublayers in a model of nested
+    layers. This is necessary for certain keras model behaviour such as displaying all layers on a
+    `model.summary(epxand_nested=True)` call.
```

Comment:
Noting this here for convenience: 

* This is also required for accessing intermediate activations. 
* Computing gradients w.r.t layers which is useful for interpretability.

Reply:
Getting access to the intermediate layers was actually the original motivation for this! I'm glad to see that someone else wants this - good motivation for me to implement and get it working 💪

Reply:
Absolutely! Please feel to ping me if I can be of any help to ship this in full glory.